vllm/v1/core/kv_cache_manager.py

```diff
@@ -22,32 +20,45 @@
 
 @dataclass
 class KVCacheBlocks:
-    blocks: list[KVCacheBlock]
+    """
+    The allocation result of KVCacheManager, work as the interface between 
+    Scheduler and KVCacheManager, to hide KVCacheManager's internal data 
+    structure from the Scheduler.
+    """
+    blocks: list[list[KVCacheBlock]]
+    """
+    blocks[i][j] refers to the i-th kv_cache_group and the j-th block of tokens.
+    We don't use block of tokens as the outer dimension because it assumes all
+    kv_cache_groups have the same number of blocks, which is true for now but 
+    will be broken if we want to give different block_size to different 
+    kv_cache_groups in the future.
+    """
 
     def __add__(self, other: "KVCacheBlocks") -> "KVCacheBlocks":
         """Adds two KVCacheBlocks instances."""
-        return KVCacheBlocks(self.blocks + other.blocks)
-
-    @classmethod
-    def create_empty(cls) -> "KVCacheBlocks":
```

Comment:
Is there a reason why this was removed? 

It's used here, so this change broke main: https://github.com/vllm-project/vllm/blob/8267f9916f9b4ceace3a8daa908d52824068862b/vllm/distributed/kv_transfer/kv_connector/v1/multi_connector.py#L163-L164

Reply:
yeah I am also looking into the same thing

Reply:
It's replaced by `kv_cache_manager.create_empty_block_list()`, because the `KVCacheBlocks` class does not know the number of kv cache groups.
Not sure why this is not detected in CI 🤔